tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/MuleClassPathClassifier.java

```diff
@@ -0,0 +1,394 @@
+                                                       .match(dependency -> dependency.isCompileScope() && !exclusion.test(dependency))
+                                       )
+                                       .collectTransitiveDependencies(
+                                               new TransitiveDependenciesFilterBuilder()
+                                                       .match(transitiveDependency -> transitiveDependency.isCompileScope() && !exclusion.test(transitiveDependency))
+                                       )).resolveDependencies().stream().filter(d -> !d.isPomType()).map(dependency -> artifactToClassPathURLResolver.resolveURL(dependency, classPathURLs)).collect(Collectors.toSet());
+        return new PluginUrlClassification(compileArtifact.getArtifactId(), Lists.newArrayList(urls));
+    }
+
+    /**
+     * The list of exclusion GAT to be excluded from application/plugin classloaders due to these are supposed to
+     * be exposed by the container.
+     * It defined by the file excluded.properties and can be changed by having this file in the module that is tested or
+     * appended to the default excluded GAT by marking the test with the annotation {@link ArtifactClassLoaderRunnerConfig}.
+     *
+     * @param klass
+     * @return a {@link Predicate} to be used in order to excluded maven artifacts from application/plugin class loaders.
+     */
+    private Predicate<MavenArtifact> getExclusionsPredicate(final Class<?> klass)
+    {
+        Predicate<MavenArtifact> exclusionPredicate = null;
+        try
+        {
+            Properties excludedProperties = getExcludedProperties();
+            String excludedModules = excludedProperties.getProperty("excluded.modules");
+            if (excludedModules != null)
+            {
+                exclusionPredicate = createPredicate(exclusionPredicate, excludedModules);
+            }
+            else
+            {
+                logger.warn("excluded.properties found but there is no list of modules defined to be excluded, this could be the reason why the test may fail later due to JUnit classes are not found");
+            }
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Error while loading excluded.properties file", e);
+        }
+        List<String> exclusionsAnnotated = getAnnotationAttributeFromHierarchy(klass, ArtifactClassLoaderRunnerConfig.class, "exclusions");
+        for (String exclusionsToBeAppended : exclusionsAnnotated)
+        {
+            if (exclusionsToBeAppended != null && exclusionsToBeAppended.length() > 0)
+            {
+                exclusionPredicate = createPredicate(exclusionPredicate, exclusionsToBeAppended);
+            }
+        }
+
+        // If no exclusion is defined the predicate should always return false to any artifact due to none is excluded
+        return exclusionPredicate == null ? new TruePredicate<MavenArtifact>().negate() : exclusionPredicate;
```

Comment:
Can't you use x -> false instead of new TruePredicate<MavenArtifact>().negate()? (and x -> true instead of new TruePredicate())